File: app/src/main/java/com/ricamgar/notify/main/MainActivity.java
package com.ricamgar.notify.main;

import android.content.Context;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;

import com.ricamgar.notify.R;
import com.ricamgar.notify.addreminder.AddReminderActivity;
import com.ricamgar.notify.reminderslist.RemindersListFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(v -> openAddReminderActivity());

        ViewPager viewPager = (ViewPager) findViewById(R.id.viewpager);
        viewPager.setAdapter(new MainAdapter(this, getSupportFragmentManager()));
        viewPager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener(){
            @Override
            public void onPageSelected(int position) {
                if (position == 0) {
                    fab.show();
                } else {
                    fab.hide();
                }
            }
        });

        TabLayout tabLayout = (TabLayout) findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);
    }

    private void openAddReminderActivity() {
        startActivity(AddReminderActivity.getIntent(this));
    }

    private static class MainAdapter extends FragmentPagerAdapter {

        private final Context context;

        public MainAdapter(Context context, FragmentManager fm) {
            super(fm);
            this.context = context;
        }

        @Override
        public Fragment getItem(int position) {
            return RemindersListFragment.newInstance(position);
        }

        @Override
        public int getCount() {
            return 2;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return position == 0 ? context.getString(R.string.tab_reminders)
                    : context.getString(R.string.tab_history);
        }
    }

}

File: app/src/main/java/com/ricamgar/notify/main/MainActivity.kt
package com.ricamgar.notify.main

import android.content.Context
import android.os.Bundle
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AppCompatActivity

import com.ricamgar.notify.R
import com.ricamgar.notify.addreminder.AddReminderActivity
import com.ricamgar.notify.reminderslist.RemindersListFragment

class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val fab = findViewById<FloatingActionButton>(R.id.fab)
    fab.setOnClickListener { openAddReminderActivity() }

    val viewPager = findViewById<ViewPager>(R.id.viewpager)
    viewPager.adapter = MainAdapter(this, supportFragmentManager)
    viewPager.addOnPageChangeListener(object : ViewPager.SimpleOnPageChangeListener() {
      override fun onPageSelected(position: Int) {
        if (position == 0) {
          fab.show()
        } else {
          fab.hide()
        }
      }
    })

    val tabLayout = findViewById<TabLayout>(R.id.tabs)
    tabLayout.setupWithViewPager(viewPager)
  }

  private fun openAddReminderActivity() {
    startActivity(AddReminderActivity.getIntent(this))
  }

  private class MainAdapter internal constructor(private val context: Context, fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment = RemindersListFragment.newInstance(position)

    override fun getCount(): Int = 2

    override fun getPageTitle(position: Int): CharSequence {
      return if (position == 0)
        context.getString(R.string.tab_reminders)
      else
        context.getString(R.string.tab_history)
    }
  }

}
